simulation/simulation.py

```diff
@@ -0,0 +1,400 @@
+            state_interaction_matrix = np.outer(state, state)
+            state_flip_probabilities = np.zeros(shape=self._settings.dynamic_state_count)
+            # Role dice to enter a dynamic state:
+            for i in range(self._settings.dynamic_state_count):
+                if state[i] == 1:
+                    continue
+                value = -6 + (state_interaction_matrix * self._dynamic_state_entry_coefs[i]).sum()
+                state_flip_probabilities[i] = logistic(value)
+            # Role dice to exit a dynamic state:
+            for i in range(self._settings.dynamic_state_count):
+                if state[i] == 0:
+                    continue
+                state_flip_probabilities[i] = self._dynamic_state_exit_probabilities[i]
+            state[:self._settings.dynamic_state_count] = np.logical_xor(state[:self._settings.dynamic_state_count],
+                                                                        np.random.binomial(n=1,
+                                                                                           p=state_flip_probabilities))
+            # Role dice to observe a concept:
+            concept_probabilities = np.zeros(shape=self._settings.concept_count)
+            for i in range(self._settings.concept_count):
+                value = -8 + (state_interaction_matrix * self._concept_emmision_coefs[i]).sum()
+                concept_probabilities[i] = logistic(value)
+            admission_concepts = (np.random.binomial(n=1, p=concept_probabilities) != 0)
+            visit = sum(admission_concepts) > 0
+            if visit:
+                # Role dice again to observe a concept (to simulate the fact that some concepts are more likely to be
+                # observed during a visit):
+                observed_concepts = admission_concepts | (np.random.binomial(n=self._settings.visit_multiplier,
+                                                                             p=concept_probabilities) != 0)
+                if is_prediction and t > index_day:
+                    prediction_labels = prediction_labels | observed_concepts
+                else:
+                    concept_ids = self._concept_ids[observed_concepts != 0]
+                    for concept_id in concept_ids:
+                        self._cdm_data.add_condition_occurrence(person_id=person_id,
+                                                                condition_start_date=start_date + t,
+                                                                condition_concept_id=concept_id,
+                                                                visit_occurrence_id=visit_occurrence_id)
+                    # If admission concept is serious, make the visit an emergency room visit, otherwise make it an
+                    # outpatient visit:
+                    if np.in1d(self._concept_ids[admission_concepts != 0], self._serious_concepts).any():
```

Comment:
This is not entirely correct, since the `concept_ids` variable also includes non-admission concepts, whereas only the admission concepts determine the visit type. 

However, I will change the code so it uses `admission_concepts`, which I'll rename to `admission_concept_idx` just to avoid confusion between concept IDs and (boolean) indexes of said concept IDs.